### PR TITLE
Device state current thingy

### PR DESCRIPTION
**Pull request by britaumich**
Add table DeviceCurrentState to store the last state of devices and rooms.
Edit check_redis_status to write the last state (create or update) into the DeviceCurrentState table.
Edit the code to use the DeviceCurrentState table to display data about rooms.

### CODE REVIEW

File: app/views/rooms/_room_assets.html.erb [hunk 1]
```diff
@@ -1,28 +1,16 @@
...
-          <div class="text-red-500 pl-2">
-            <%= "#{state.key} - #{state.value}" %>
-          </div>
-        <% else %>
-          <div class="pl-2">
-            <%= "#{state.key} - #{state.value}" unless state.key == "Error State" %>
-          </div>
-        <% end %>
+    <div class="font-bold"><%= device.name %></div>
+    <% DeviceCurrentState.where(device_id: device.id).each do |state| %>
```
Comment (rsmoke) [inline app/views/rooms/_room_assets.html.erb:4]: We should move this call to active record out of the view and into the controller

File: app/views/rooms/_room_state.html.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div class="text-xs px-2">
-  <% DeviceState.where(device_id: device.id).select(:key, :value, 'MAX(created_at)').group(:key, :value).each do |state| %>
+  <% DeviceCurrentState.where(device_id: device.id).each do |state| %>
```
Comment (rsmoke) [inline app/views/rooms/_room_state.html.erb:2]: We should move these calls to active record out of the view and into the controller

File: lib/tasks/check_redis_status.rake [hunk 3]
```diff
@@ -13,6 +12,40 @@ task check_redis_status: :environment do
   require 'redis'
   redis = Redis.new(host: "localhost")
 
+  def write_socket_data_to_db(room, h)
+    h["LSARoom"].each do |input, data|
```
Comment (rsmoke) [inline lib/tasks/check_redis_status.rake:16]: use a more descriptive variable name for when this is code is looked at  by any developer it is clear and there isn't a need to fish around the code base to figure stuff out

**Review by rsmoke — CHANGES_REQUESTED** [review]
I left inline comments for moving logic our of views and into controllers, creating more descriptive variable names. Also the check_redis_status/rake task conditional can better be done with method calls.

File: app/views/rooms/_room_assets.html.erb (hunk 1 above)
Comment (britaumich) [reply app/views/rooms/_room_assets.html.erb]: _room_assets partial is rendered in _room_card partial with {devices: get_room_asset_devices(room) } argument.
I can create another helpers method (get_current_states_for_device(device)) to make it look prettier in the view. But I don't think I can't move this call to active record into the controller. The rooms controller sends @rooms to the index view. Can it send all devices and device_sates of all rooms too?

File: app/views/rooms/_room_state.html.erb (hunk 2 above)
Comment (britaumich) [reply app/views/rooms/_room_state.html.erb]: This file is not needed. I removed it as well as two unused helper methods.

File: lib/tasks/check_redis_status.rake (hunk 3 above)
Comment (britaumich) [reply lib/tasks/check_redis_status.rake]: Sorry about that. I replayed 'h' with 'payload'.

**Review by rsmoke — APPROVED** [review]
Thank you Rita! your  code pretty close to being perfect. IF you can get conditionals reined in.
🍻 🌭 🕺

**Pull request merged**